File: PROGRAMMERS/LEVEL1/체육복.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n, vector<int> lost, vector<int> reserve) {
    int answer = 0;
    int arr[30];
    for (int i = 0; i < n; i++) { arr[i] = 1; }
    for (int i = 0; i < lost.size(); i++) { arr[lost[i] - 1]--; }
    for (int i = 0; i < reserve.size(); i++) { arr[reserve[i] - 1]++; }

    for (int i = 0; i < n; i++) {
        if (arr[i] == 2 && arr[i - 1] == 0 && i >= 1) {
            arr[i]--;
            arr[i - 1]++;
        }
        if (arr[i] == 2 && arr[i + 1] == 0 && i < n) {
            arr[i]--;
            arr[i + 1]++;
        }
    }
    for (int i = 0; i < n; i++) {
        if (arr[i] > 0) answer++;
    }

    return answer;
}

File: PROGRAMMERS/LEVEL1/체육복.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        int answer = 0;
        int[] student = new int[30];
        
        Arrays.fill(student, 1);
        for(int i=0;i<lost.length;i++) student[lost[i]-1]--;
        for(int i=0;i<reserve.length;i++) student[reserve[i]-1]++;
        
        for(int i=0;i<n;i++) {
        	if(student[i]==2 && i>0 &&student[i-1]==0) {
        		student[i]--;
        		student[i-1]++;
        	}
        	if(student[i]==2 && i<n-1 && student[i+1]==0) {
        		student[i]--;
        		student[i+1]++;
        	}
        }
        for(int i=0;i<n;i++) 
        	if(student[i]>0)
        		answer++;
        
        return answer;
    }
}